proxy-router/internal/config/models_config.go

```diff
@@ -0,0 +1,58 @@
+package config
+
+import (
+	"encoding/json"
+
+	"github.com/MorpheusAIs/Morpheus-Lumerin-Node/proxy-router/internal/lib"
+)
+
+type ModelConfigLoader struct {
+	log *lib.Logger
+}
+
+type ModelConfig struct {
+	ModelName string `json:"modelName"`
+	ApiType   string `json:"apiType"`
+	ApiURL    string `json:"apiUrl"`
+	ApiKey    string `json:"apiKey"`
+}
+
+type ModelConfigs map[string]ModelConfig
+
+func NewModelConfigLoader(log *lib.Logger) *ModelConfigLoader {
+	return &ModelConfigLoader{
+		log: log,
+	}
+}
+
+func (e *ModelConfigLoader) ModelConfigFromID(ID string) *ModelConfig {
```

Comment:
I suggest doing it only once, on startup 

func (e *ModelConfigLoader) Init() error

and then just using mapping. This way you validate the config file early and avoid repeatably parsing config file